lib/puppet/parser/functions/validate_bool.rb

```diff
@@ -1,41 +0,0 @@
-# frozen_string_literal: true
-
-#
-# validate_bool.rb
-#
-module Puppet::Parser::Functions
-  newfunction(:validate_bool, doc: <<-DOC
-    @summary
-      Validate that all passed values are either true or false. Abort catalog
-      compilation if any value fails this check.
-
-    @return
-      validate boolean
-
-    @example **Usage**
-
-      The following values will pass:
-
-          $iamtrue = true
-          validate_bool(true)
-          validate_bool(true, true, false, $iamtrue)
-
-      The following values will fail, causing compilation to abort:
-
-          $some_array = [ true ]
-          validate_bool("false")
-          validate_bool("true")
-          validate_bool($some_array)
-      DOC
-  ) do |args|
```

Comment:
I dont see any mention of deprecation in this function. Could this Parser Function deletion be a mistake?

Reply:
The `validate_bool()` function indeed does not produce deprecation message, but it call `is_bool()` on line 36 which does. I assumed no deprecation message was put here in order to avoid some kind of "duplicate deprecation warning".

Reply:
Ah I see. Yeah, that looks to be correct.